### PR TITLE
Fix tx execution bug, and improve logging

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why is this change needed?

Transactions with too little gas kill the enclave. 
Logging needs to also be a bit better to help diagnose issues

### What changes were made as part of this PR:

1. Fix a bug where an account was credited even though the transaction that triggered that crediting would fail. 
2. Improve some log messages, and standardise them
3. Previously all transactions that errored were excluded which is not the right behaviour. Now only transactions that fail because the nonce was too high are excluded.
4. Disable the "ifshort" lint check because it's broken.

### What are the key areas to look at
- The evm_facade

### CODE REVIEW

File: go/enclave/rollupchain/rollup_chain.go [hunk 1]
```diff
@@ -294,23 +315,37 @@ func (rc *RollupChain) processState(rollup *obscurocore.Rollup, txs obscurocore.
 		stateDB,
 	)
 
-	depositReceipts := evm.ExecuteTransactions(depositTxs, stateDB, rollup.Header, rc.storage, rc.obscuroChainID, len(successfulTransactions))
-	if len(depositReceipts) != len(depositTxs) {
+	depositResponses := evm.ExecuteTransactions(depositTxs, stateDB, rollup.Header, rc.storage, rc.obscuroChainID, len(executedTransactions))
+	depositReceipts := make([]*types.Receipt, len(depositResponses))
```
Comment (joeldudley) [inline go/enclave/rollupchain/rollup_chain.go:319]: Shouldn't this come after the check on the next line?

**Review by joeldudley — APPROVED** [review]
LGTM, with some very minor style comments. I can't say I understand it fully, but it looks reasonable.

File: go/enclave/rollupchain/rollup_chain.go [hunk 2]
```diff
@@ -294,23 +315,37 @@ func (rc *RollupChain) processState(rollup *obscurocore.Rollup, txs obscurocore.
...
+	if len(depositResponses) != len(depositTxs) {
 		log.Panic("Sanity check. Some deposit transactions failed.")
 	}
+	i := 0
+	for _, resp := range depositResponses {
```
[the remark this answers is not in the document]
Comment (tudor-malene) [reply go/enclave/rollupchain/rollup_chain.go:324]: can you? this is iterating over a map

[the remark this answers is not in the document]
Comment (joeldudley) [reply go/enclave/rollupchain/rollup_chain.go:324]: Oops. You're the boss :)

**Pull request merged**